### PR TITLE
EXSWHTEC-3 - Implement tests for hipDeviceReset

### PR DESCRIPTION
**Pull request by mirza-halilcevic**
- Basic test for behaviour validation
- Multithreaded test similar to basic test.

### CODE REVIEW

**Review by chrispaquot — DISMISSED** [review]

**Review by chrispaquot — COMMENTED** [review]
You should also test that an allocated memory and stream aren't valid anymore after device reset.

**Comment by mirza-halilcevic** [comment]
> You should also test that an allocated memory and stream aren't valid anymore after device reset.

@chrispaquot The suggested checks are implemented. Please proceed with the review.

**Review by chrispaquot — DISMISSED** [review]

File: tests/catch/unit/device/hipDeviceReset.cc [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+  unsigned int flags_before = 0u;
+  HIP_CHECK(hipGetDeviceFlags(&flags_before));
+  hipSharedMemConfig mem_config_before;
+  HIP_CHECK(hipDeviceGetSharedMemConfig(&mem_config_before));
+
+  void* ptr = nullptr;
+  HIP_CHECK(hipMalloc(&ptr, 500));
+  hipStream_t stream = nullptr;
+  HIP_CHECK(hipStreamCreate(&stream));
+#if HT_NVIDIA
```
Comment (chrispaquot) [inline tests/catch/unit/device/hipDeviceReset.cc]: why those?

Comment (mirza-halilcevic) [reply tests/catch/unit/device/hipDeviceReset.cc]: hipDeviceSetCacheConfig is not supported on AMD and returns hipErrorNotSupported.

Comment (mangupta) [reply tests/catch/unit/device/hipDeviceReset.cc]: I would rather we relax the code to allow hipSuccess or hipErrorNotSupported.

Comment (chrispaquot) [reply tests/catch/unit/device/hipDeviceReset.cc]: Agreed with Maneesh. Or if the error code is hipErrorNotSupported then don't enforce checks on hipDeviceSetCacheConfig

Comment (mirza-halilcevic) [reply tests/catch/unit/device/hipDeviceReset.cc]: Done. You can proceed with the review.

**Review by chrispaquot — DISMISSED** [review]

**Comment by mangupta** [comment]
@music-dino / @mirza-halilcevic : Some tests failed on CI. Details shared separately.

**Comment by mirza-halilcevic** [comment]
> @music-dino / @mirza-halilcevic : Some tests failed on CI. Details shared separately.

PR updated, you can proceed with the merge.

**Review by chrispaquot — DISMISSED** [review]

**Review by gargrahul — APPROVED** [review]

**Pull request merged**